packages/permissionless/accounts/kernel/signerToEcdsaKernelSmartAccount.ts

```diff
@@ -73,52 +89,175 @@ const createAccountAbi = [
+ * Default addresses map for different kernel smart account versions
  */
-const KERNEL_ADDRESSES: {
-    ECDSA_VALIDATOR: Address
-    ACCOUNT_V2_2_LOGIC: Address
-    FACTORY_ADDRESS: Address
+export const KERNEL_VERSION_TO_ADDRESSES_MAP: {
+    [key in KernelVersion]: {
+        ECDSA_VALIDATOR: Address
+        ACCOUNT_LOGIC: Address
+        FACTORY_ADDRESS: Address
+        META_FACTORY_ADDRESS?: Address
+    }
 } = {
-    ECDSA_VALIDATOR: "0xd9AB5096a832b9ce79914329DAEE236f8Eea0390",
-    ACCOUNT_V2_2_LOGIC: "0x0DA6a956B9488eD4dd761E59f52FDc6c8068E6B5",
-    FACTORY_ADDRESS: "0x5de4839a76cf55d0c90e2061ef4386d962E15ae3"
+    "0.2.2": {
+        ECDSA_VALIDATOR: "0xd9AB5096a832b9ce79914329DAEE236f8Eea0390",
+        ACCOUNT_LOGIC: "0x0DA6a956B9488eD4dd761E59f52FDc6c8068E6B5",
+        FACTORY_ADDRESS: "0x5de4839a76cf55d0c90e2061ef4386d962E15ae3"
+    },
+    "0.3.0-beta": {
+        ECDSA_VALIDATOR: "0x8104e3Ad430EA6d354d013A6789fDFc71E671c43",
+        ACCOUNT_LOGIC: "0x94F097E1ebEB4ecA3AAE54cabb08905B239A7D27",
+        FACTORY_ADDRESS: "0x6723b44Abeec4E71eBE3232BD5B455805baDD22f",
+        META_FACTORY_ADDRESS: "0xd703aaE79538628d27099B8c4f621bE4CCd142d5"
+    }
+}
+
+/**
+ * Get supported Kernel Smart Account version based on entryPoint
+ * @param entryPoint
+ */
+const getKernelVersion = (entryPoint: EntryPoint): KernelVersion => {
+    return entryPoint === "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"
+        ? "0.2.2"
+        : "0.3.0-beta"
+}
+
+type KERNEL_ADDRESSES = {
+    ecdsaValidatorAddress: Address
+    accountLogicAddress: Address
+    factoryAddress: Address
+    metaFactoryAddress: Address
+}
+
+/**
+ * Get default addresses for Kernel Smart Account based on entryPoint or user input
+ * @param entryPointAddress
+ * @param ecdsaValidatorAddress
+ * @param accountLogicAddress
+ * @param factoryAddress
+ * @param metaFactoryAddress
+ */
+const getDefaultAddresses = <entryPoint extends EntryPoint>(
+    entryPointAddress: entryPoint,
+    {
+        ecdsaValidatorAddress: _ecdsaValidatorAddress,
+        accountLogicAddress: _accountLogicAddress,
+        factoryAddress: _factoryAddress,
+        metaFactoryAddress: _metaFactoryAddress
+    }: Partial<KERNEL_ADDRESSES>
+): KERNEL_ADDRESSES => {
+    const kernelVersion = getKernelVersion(entryPointAddress)
+    const addresses = KERNEL_VERSION_TO_ADDRESSES_MAP[kernelVersion]
+    const ecdsaValidatorAddress =
+        _ecdsaValidatorAddress ?? addresses.ECDSA_VALIDATOR
+    const accountLogicAddress = _accountLogicAddress ?? addresses.ACCOUNT_LOGIC
+    const factoryAddress = _factoryAddress ?? addresses.FACTORY_ADDRESS
+    const metaFactoryAddress =
+        _metaFactoryAddress ?? addresses?.META_FACTORY_ADDRESS ?? zeroAddress // Meta Factory doesn't exists for Kernel v2.2
+
+    return {
+        ecdsaValidatorAddress,
+        accountLogicAddress,
+        factoryAddress,
+        metaFactoryAddress
+    }
+}
+
+export const getEcdsaRootIdentifierForKernelV3 = () => {
```

Comment:
this is hard coding the ECDSA_VALIDATOR address, but we hare taking this address as a param in `signerTo..*`. We should use that address as it also defaults to `KERNEL_VERSION_TO_ADDRESSES_MAP["0.3.0-beta"].ECDSA_VALIDATOR`

Reply:
Makes sense.